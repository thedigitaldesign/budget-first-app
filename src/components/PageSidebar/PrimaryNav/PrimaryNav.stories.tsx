import React from 'react'

// Packages
import { storiesOf } from '@storybook/react'

// Components
import PrimaryNav from './PrimaryNav'

storiesOf('Components/PrimaryNav', module).add('Primary Nav', () => {
  return (
    <div>
      <PrimaryNav />
    </div>
  )
})
