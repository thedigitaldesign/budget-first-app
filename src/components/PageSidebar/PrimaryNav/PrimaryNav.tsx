import React, { Component } from 'react'

// Axios
import axios from 'axios'

export default class PrimaryNav extends Component {
  componentDidMount() {
    axios.get('./nav.json').then(response => {
      response.data.nav
    })
  }

  render() {
    return (
      <nav className="primary-nav">
        <div></div>
      </nav>
    )
  }
}
