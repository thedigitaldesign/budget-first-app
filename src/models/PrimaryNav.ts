export interface PrimaryNav {
  title: string
  icon?: string
  items: [
    {
      title: string
      href: string
    },
  ]
}
