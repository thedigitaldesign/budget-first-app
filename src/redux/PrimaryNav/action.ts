// Packages
import axios from 'axios'

// Dispatches
import * as type from '../Types'

// export class PrimaryNavAction {
  export const GetPrimaryNav = () => {
    return (dispatch: any) => {
      dispatch(FetchPrimaryNavRequest())

      axios
        .get('./nav.json')
        .then(response => {
          const nav = response.data
          dispatch(FetchPrimaryNavSuccess(nav))
        })
        .catch(error => {
          dispatch(FetchPrimaryNavFailure(error.message))
        })
    }
  }

  export const FetchPrimaryNavRequest = () => {
    return {
      type: type.PrimaryNav.FETCH_NAV_REQUEST,
    }
  }

  export const FetchPrimaryNavSuccess = (nav: [{}]) => {
    return {
      type: type.PrimaryNav.FETCH_NAV_SUCCESS,
      payload: nav,
    }
  }

  export const FetchPrimaryNavFailure = (error: any) => {
    return {
      type: type.PrimaryNav.FETCH_NAV_FAILURE,
      payload: error,
    }
  }
// }
