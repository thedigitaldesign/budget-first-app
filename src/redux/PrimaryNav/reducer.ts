import { PrimaryNav as type } from '../Types'

const initialState = {
  loading: false,
  PrimaryNav: [{}],
  error: '',
}

interface action {
  type: string
  payload: any
}

export const PrimaryNavReducer = (state = initialState, action: action) => {
  switch(action.type) {
    case type.FETCH_NAV_REQUEST:
      return {
        ...state,
        loading: true
      }
    case type.FETCH_NAV_SUCCESS:
      return {
        loading: false,
        PrimaryNav: action.payload,
        error: ''
      }
    case type.FETCH_NAV_FAILURE:
      return {
        loading: false,
        PrimaryNav: [{}],
        error: action.payload
      }
    default: return state
  }
}

// export default reducer
