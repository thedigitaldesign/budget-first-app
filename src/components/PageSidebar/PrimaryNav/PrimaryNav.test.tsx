import React from 'react'

// Packages
import { shallow } from 'enzyme'
import axios from 'axios'

// Components
import PrimaryNav from './PrimaryNav'

describe('PrimaryNav', () => {
  const logo = shallow(<PrimaryNav />)

  it('renders correctly', () => {
    expect(logo).toMatchSnapshot()
  })

  describe('axios', () => {
    jest.mock('axios')

    // it('returns a json object for the nav', async () => {
    //   const data = {
    //     nav: [
    //       {
    //         title: 'Dashboard',
    //         icon: 'fal fa-info-circle',
    //         items: [
    //           {
    //             title: 'Dashboard',
    //             href: 'dashboard',
    //           },
    //         ],
    //       },
    //     ],
    //   }
  
    //   axios.get.mockImplementationOnce(() => Promise.resolve(data))

    //   await expect(fetchData('react')).resolves.toEqual(data)
    // })
  })

  
})
