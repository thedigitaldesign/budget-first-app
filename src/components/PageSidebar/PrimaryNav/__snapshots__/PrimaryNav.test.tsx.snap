// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryNav renders correctly 1`] = `ShallowWrapper {}`;
